synapse/http/server.py

```diff
@@ -223,6 +301,9 @@ def register_paths(
         If the regex contains groups these gets passed to the callback via
         an unpacked tuple.
 
+        The callback may be marked with the `@cancellable` decorator, which will
+        cause request processing to be cancelled when clients disconnect early.
+
```

Comment:
I wonder if we should add an explicit `cancellable` parameter to `register_paths` instead.
There are a few places where we call `register_paths` manually, with callbacks that don't match the `^on_(GET|PUT|POST|DELETE)$` pattern, eg.

https://github.com/matrix-org/synapse/blob/4bc8cb4669ddeb719a3a6de39b093fc3be8db6fe/synapse/rest/client/room.py#L133-L138

There we can either relax the validation in the `@cancellable` decorator or add an explicit `cancellable` parameter to `register_paths`.